Adding map server raw mode implemented by DLu

aaronhoy opened the pull request:
Originally implemented by DLu (and I've added some minor cleanup), this adds a mode to the map server that simply passes pixel values straight through, allowing more flexibility for how maps are used.

DLu committed:
Use enum to control map interpretation

aaronhoy committed:
Removing some trailing whitespace.

aaronhoy commented:
I tested both the new mode and the default behavior (trinary mode) and everything seems to be working as expected.

mikeferguson commented on map_server/src/image_loader.cpp:
```diff
@@ -99,7 +99,7 @@ loadMapFromFile(nav_msgs::GetMap::Response* resp,
   n_channels = img->format->BytesPerPixel;
 
-  if (trinary || n_channels == 1)
+  if (n_channels <= 3)
```
are we sure on this change?

mikeferguson replied:
I think this should actually be "mode==Trinary || n_channels == 1" -- I'm not even sure why mode==Trinary is there, but let's not change the functionality of trinary mode

aaronhoy committed:
Corrections to alpha channel detection and usage.

Changing to actually detect whether the image has an alpha channel instead of
inferring based on the number of channels.
Also reverting to legacy behavior of trinary mode overriding alpha removal.
This will cause the alpha channel to be averaged in with the others in trinary
mode, which is the current behavior before this PR.

aaronhoy commented:
Correction pushed to address comment from @mikeferguson .
The change now properly checks for the presence of an alpha channel, but still allows trinary mode to override that so as to preserve legacy behavior.

mikeferguson commented:
@aaronhoy can you open a similar PR against jade-devel

aaronhoy commented:
@mikeferguson, I opened #428

DLu commented:
Thanks for following up on this @aaronhoy, and for catching my bugs.
